pkg/controller/poolcoordinator/cert/poolcoordinator_cert_manager.go

```diff
@@ -0,0 +1,165 @@
+	poolcoordinatorNS = "kube-system"
+	// poolcoordinatorAPIServerSVC = "pool-coordinator-apiserver"
+	poolcoordinatorAPIServerSVC        = "kube-dns" // for test
+	poolcoordinatorAPIServerSecertName = "pool-coordinator-apiserver-certs"
+	poolcoordinatorETCDSecertName      = "pool-coordinator-etcd-certs"
+	commonNamePrefix                   = "system:node:poolcoordinator"
+)
+
+// PoolCoordinatorCertManager manages certificates releted with poolcoordinator pod
+type PoolCoordinatorCertManager struct {
+	kubeclientset client.Interface
+	podLister     corelisters.PodLister
+	podSynced     cache.InformerSynced
+	podWorkQueue  workqueue.RateLimitingInterface
+}
+
+func NewPoolCoordinatorCertManager(kc client.Interface, podInformer coreinformers.PodInformer) *PoolCoordinatorCertManager {
+
+	certManager := PoolCoordinatorCertManager{
+		kubeclientset: kc,
+
+		podLister: podInformer.Lister(),
+		podSynced: podInformer.Informer().HasSynced,
+
+		podWorkQueue: workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+	}
+
+	// Watch for poolcoordinator pod changes to manage related certs (including kubeconfig)
+	podInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(obj interface{}) {},
+		UpdateFunc: func(oldObj, newObj interface{}) {},
+		DeleteFunc: func(obj interface{}) {},
+	})
+
+	return &certManager
+}
+
+func (c *PoolCoordinatorCertManager) Run(threadiness int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+
+	klog.Info("Starting poolcoordinatorCertManager controller")
+	defer klog.Info("Shutting down poolcoordinatorCertManager controller")
+	defer c.podWorkQueue.ShutDown()
+
+	// Prepare certs shared among all poolcoordinators
+	if err := initPoolCoordinatorCerts(c.kubeclientset, stopCh); err != nil {
+		klog.Errorf("poolcoordinatorCertManager init certs fail, %v", err)
+	}
+
+	// Synchronize the cache before starting to process events
+	if !cache.WaitForCacheSync(stopCh, c.podSynced) {
+		klog.Error("sync poolcoordinatorCertManager controller timeout")
+	}
+
+	// The main Controller loop
+	for i := 0; i < threadiness; i++ {
+		go wait.Until(c.runWorker, time.Second, stopCh)
+	}
+
+	<-stopCh
+}
+
+func (c *PoolCoordinatorCertManager) runWorker() {
+	for {
+		obj, shutdown := c.podWorkQueue.Get()
+		if shutdown {
+			return
+		}
+
+		if err := c.syncHandler(obj.(string)); err != nil {
+			utilruntime.HandleError(err)
+		}
+
+		c.podWorkQueue.Forget(obj)
+		c.podWorkQueue.Done(obj)
+	}
+}
+
+func (c *PoolCoordinatorCertManager) syncHandler(key string) error {
+	return nil
+}
+
+func initPoolCoordinatorCerts(clientSet client.Interface, stopCh <-chan struct{}) error {
+
+	// 1. get poolcoordinator svc IP
+	// Notice: poolcoordinator svc should be installed properly before yurt-controller-manager
+	apiServerSVC, err := clientSet.CoreV1().Services(poolcoordinatorNS).Get(context.TODO(), poolcoordinatorAPIServerSVC, v1.GetOptions{})
```

Comment:
please wait until the service `poolcoordinatorAPIServerSVC` is ready.

Reply:
done